pybamm/pybamm_data.py

```diff
@@ -0,0 +1,137 @@
+import pooch
+import pathlib
+
+
+class DataLoader:
+    """
+    Data Loader class for downloading and loading data files upstream at https://github.com/pybamm-team/pybamm-data/
+
+    Attributes
+    ----------
+    path: POSIX/Windows PATH
+       POSIX/Windows path to the data file.
+       The folder locations are defined by the `platformdirs <https://platformdirs.readthedocs.io/en/latest/>`_ package using the `user_cache_dir <https://platformdirs.readthedocs.io/en/latest/api.html#cache-directory>`_ function. The default cache paths for pybamm are located under -
+
+        - Mac: ``~/Library/Caches/v1.x.x/pybamm``
+
+        - Unix: ``~/.cache/v1.x.x/pybamm`` or the value of the XDG_CACHE_HOME environment variable, if defined.
+
+        - Windows: ``C:\\Users\\<user>\\AppData\\Local\\pybamm\\v1.x.x\\Cache``
```

Comment:
```suggestion
       The folder locations are defined by the `platformdirs <https://platformdirs.readthedocs.io/en/latest/>`_ package using the `user_cache_dir <https://platformdirs.readthedocs.io/en/latest/api.html#cache-directory>`_ function.
```

I would suggest skipping these, it directs too much emphasis on users who do not have to deal with this information when using the `DataLoader` class, and therefore we can keep this abstracted away from them.